test/test_integration.py

```diff
@@ -1494,6 +1494,99 @@ def render_content(click, prev_graph):
         self.snapshot('render-empty-graph')
 
+    def test_graph_extend_trace(self):
+        app = dash.Dash(__name__)
+
+        app.layout = html.Div([
+            dcc.Graph(
+                id='trace_will_extend',
+                figure=dict(
+                    data=[{'x': [0, 1, 2, 3, 4],
+                           'y': [0, .5, 1, .5, 0]
+                           }],
+                )
+            ),
+            dcc.Graph(
+                id='trace_will_extend_selectively',
+                figure=dict(
+                    data=[{'x': [0, 1, 2, 3, 4],
+                           'y': [0, .5, 1, .5, 0]
+                           },
+                          {'x': [0, 1, 2, 3, 4],
+                           'y': [1, 1, 1, 1, 1]
+                           }],
+                )
+            ),
+            html.Div(id='output1'),
+            html.Div(id='output2'),
+            dcc.Interval(
+                id='interval_extendablegraph_update',
+                interval=10,
+                n_intervals=0,
+                max_intervals=1)
+        ])
+
+        @app.callback(Output('trace_will_extend', 'extendData'),
+                      [Input('interval_extendablegraph_update', 'n_intervals')])
+        def trace_will_extend(n_intervals):
+            if n_intervals is None or n_intervals < 1:
+                raise PreventUpdate
+
+            x_new = [5, 6, 7, 8, 9]
+            y_new = [.1, .2, .3, .4, .5]
+            return dict(x=[x_new], y=[y_new]), [0]
+
+        @app.callback(Output('trace_will_extend_selectively', 'extendData'),
+                      [Input('interval_extendablegraph_update', 'n_intervals')])
+        def trace_will_extend_selectively(n_intervals):
+            if n_intervals is None or n_intervals < 1:
+                raise PreventUpdate
+
+            x_new = [5, 6, 7, 8, 9]
+            y_new = [.1, .2, .3, .4, .5]
+            return dict(x=[x_new], y=[y_new]), [1]
+
+        @app.callback(Output('output1', 'children'),
+                      [Input('trace_will_extend', 'extendData')],
+                      [State('trace_will_extend', 'figure')])
+        def display_data1(trigger, figure):
+            return json.dumps(figure['data'])
+
+        @app.callback(Output('output2', 'children'),
+                      [Input('trace_will_extend_selectively', 'extendData')],
+                      [State('trace_will_extend_selectively', 'figure')])
+        def display_data2(trigger, figure):
+            return json.dumps(figure['data'])
+
+        self.startServer(app)
+
+        try:
+            comparison = json.dumps([
+                dict(
+                    x=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
+                    y=[0, .5, 1, .5, 0, .1, .2, .3, .4, .5]
+                )
+            ])
+            self.wait_for_text_to_equal('#output1', comparison)
+        except TimeoutException:
+            raise Exception('Graph data was not extended.')
```

Comment:
Let's not wrap these in an extra `try/except` - let the error in `wait_for_text_to_equal` do its own reporting. It looks like the version of that function in this repo isn't as friendly as some of the others, like the one in `dash-renderer` that will tell you what text was found instead when the wait timed out. But we have a separate effort under way right now to DRY those (cc @T4rk1n @byronz) so I wouldn't worry about improving it in this PR.